src/pagination/dates.rs

```diff
@@ -0,0 +1,159 @@
+use chrono::Datelike;
+use chrono::Timelike;
+
+use crate::cobalt_model::pagination_config::DateIndex;
+
+use crate::cobalt_model::DateTime;
+use crate::document::Document;
+
+use super::*;
+use helpers::extract_scalar;
+use paginator::Paginator;
+
+#[derive(Debug)]
+struct DateIndexHolder<'a> {
+    value: u32,
+    field: Option<DateIndex>,
+    posts: Vec<&'a liquid::value::Value>,
+    sub_date: Vec<DateIndexHolder<'a>>,
+}
+
+impl<'a> DateIndexHolder<'a> {
+    fn new(value: u32, field: Option<DateIndex>) -> Self {
+        DateIndexHolder {
+            value,
+            field,
+            posts: vec![],
+            sub_date: vec![],
+        }
+    }
+}
+
+fn extract_published_date<'a>(value: &'a liquid::value::Value) -> Option<DateTime> {
+    if let Some(published_date) = extract_scalar(&value, "published_date") {
+        DateTime::parse(published_date.to_str())
```

Comment:
Why are we parsing it?  Shouldn't we call [`to_date`](https://github.com/cobalt-org/liquid-rust/blob/master/liquid-value/src/scalar.rs#L113)?

Reply:
Because I didn't know its existence ^^ gonna change that